### DISCUSSION TITLE
how to change cmake build type library install path

### DISCUSSION THREAD

**Discussion by zhanlangorz** [keplerproject/luarocks#1438] [Q&A]
When i write a rockspec for librdkafka,  like this:

```
package = "librdkafka"
version = "1.9.2-1"
source = { 
   url ="xxx",
}
description = { 
   summary = "librdkafka for luarocks",
   detailed = "", 
   homepage = "https://github.com/edenhill/librdkafka",
}
dependencies = { 
   "lua >= 5.1"
}
build = { 
   type = "cmake",
   variables = { 
     CMAKE_BUILD_TYPE = "Release",
     CMAKE_INSTALL_LIBDIR = "$(PREFIX)",
     CMAKE_INSTALL_BINDIR = "$(PREFIX)",
     CMAKE_INSTALL_INCLUDEDIR = "$(PREFIX)",
   },  
}
```

Then i install this rock use  --tree option: 
`luarocks install librdkafka-1.9.2-1.src.rock --tree=./luarocks_mod`

this rock will install to luarocks_mod/lib/luarocks/rocks-5.1/librdkafka/1.9.2-1/,   i can't require this library without  the full path.

how to install the compilerd library to [--treepath]/lib/   ?

**Comment by hishamhm** [comment]
LuaRocks is not intended for installing C libraries into system paths, it is a package manager for Lua modules. You should build/install this library through other means.
